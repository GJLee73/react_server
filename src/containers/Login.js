/**
 * 로그인 컨테이너
 * @author G1
 * @logs // 18.2.25
 */

import React from 'react';
import {LoginComponent} from '../components/auth';
import {connect } from 'react-redux';
import {loginRequest,cleanCurrentUser} from '../modules/authentication';
//import {browserHistory} from 'react-router';

class Login extends React.Component{
	constructor(props){
		super(props);
		this.handleLogin = this.handleLogin.bind(this);
	}
/**
 * 로그인을 한다.
 * @param {string} id 
 * @param {string} pw 
 * @description
 * 	만약 로그인시 이메일 인증되지 않았으면 verify로 리다이렉트 한다.
 *  인증되었으면 home으로
 *  인증이 안됬으면 다시 한다.
 */
	handleLogin(id, pw){
		return this.props.loginRequest(id, pw).then(
			()=>{
				
				if(this.props.status === "SUCCESS"){
					if(!this.props.verified){
					
						this.props.history.push('/verify')

					return 0;
				}


					window.Materialize.toast('Welcome, ' + id + '!', 2000);
					this.props.history.push('/');
					return 0;
				} else{
					let $toastContent = window.$('<span style="color: $FFB4BA">Incorrect username or password</span>');
					window.Materialize.toast($toastContent, 2000);
					return false;
				}
			}
		)
	}

	render(){
		return (
			<div>
				<LoginComponent	onLogin={this.handleLogin}/>
			</div>
		);
    }	
}

const mapStateToProps = (state) => {
	return {
		status: state.authentication.getIn(['login','status']),
		verified: state.authentication.getIn(['status','verified'])
	};
};

const mapDispatchToProps = (dispatch) => {
	return {
		loginRequest: (id, pw)=>{
			return dispatch(loginRequest(id,pw));
		},
		cleanCurrentUser:()=>{
			return dispatch(cleanCurrentUser());
		}
	};
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);