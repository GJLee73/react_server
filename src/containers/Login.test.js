import React from 'react';
import {shallow} from 'enzyme'
import {Login} from './Login'

describe('Login test',()=>{
    var loginRequest = jest.fn(()=>Promise.resolve())
    var wrapper = shallow(<Login loginRequest={loginRequest}/>)
    var history = {push:jest.fn()}
    global.Materialize = {toast:jest.fn()}
    describe('handleLogin test',()=>{
        it('로그인에 성공하면 홈으로 리다이렉트 된다. ',async()=>{
            wrapper.setProps({status:"SUCCESS",history,verified:true})
           await wrapper.instance().handleLogin()
            expect(history.push).toHaveBeenCalledTimes(1)
        })
        it('아이디는 존재하나 verified==false일때 /verify로 리다이렉트 된다.',async()=>{
            wrapper.setProps({verified:false})
           await wrapper.instance().handleLogin()
            expect(history.push.mock.calls[1][0]).toBe('/verify')
        })
        it('로그인 실패시 실패 창을 띄워준다.',async()=>{
            wrapper.setProps({status:'FAILURE'})
            await wrapper.instance().handleLogin()
            expect(global.Materialize.toast).toHaveBeenCalledTimes(2)
        })

    })
    it('snapshot test',()=>{
        expect(wrapper).toMatchSnapshot()
    })
})