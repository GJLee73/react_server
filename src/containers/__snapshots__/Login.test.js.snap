// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login test snapshot test 1`] = `
<div>
  <LoginComponent
    onLogin={[Function]}
  />
</div>
`;
